lib/inspec/file_provider.rb

```diff
@@ -107,16 +108,24 @@ def initialize(gem_info)
       # Determine a path to the gem installation directory
       gem_name = gem_info[:gem]
       bootstrap_file = gem_info[:path]
+      gem_version = gem_info[:version]
       if bootstrap_file
         # We were given an explicit path - go two directories up
         parts = Pathname(gem_info[:path]).each_filename.to_a
         @gem_lib_root = File.join(parts.slice(0, parts.length - 2))
+        @files = Dir[File.join(Shellwords.shellescape(@gem_lib_root), "**", "*")]
       else
         # Look up where the gem is installed, respecting version
-        raise NotImplementedError, "TODO: Gem FileProvider when gem #{gem_name} is not a local reference"
+        loader = Inspec::Plugin::V2::Loader.new
+        gem_path = loader.find_gem_directory(gem_name, gem_version)
+        if File.exist?(gem_path)
+          gem = DirProvider.new(gem_path)
+          @files = gem.files
+          gem
+        else
+          raise "Dependency does not exist #{gem_name}"
```

Comment:
We should not raise error here.. instead return nil or a custom exception which should be captured by GemFetcher to try installing the gem ( can be handled in a different PR )

Reply:
I feel the job of gem provider is to just return the gem if it exists else just return the error. The logic to retry the installation and everything is handled in cache_fetcher, which anyway calls gem fetcher to do that. So GemProvider library is just doing a dummy job. That is what I think looking into different providers logic.

Reply:
I disagree with the scope of Gem Provider. But yeah with the current plugin system we don't follow SRP principle so okay with the scope of current gem provider, but all am asking is instead of raising a StandardError, could you raise a custom error object so we can capture it at the calling function and fallback to installation.

Reply:
Okay, let me check. If you already know, can you tell me which custom error class that is? Or are you suggesting to create a new one?

Reply:
@sathish-progress Made the changes for this.